### PR TITLE
Fix ResourceIterator and add test

### PR DESCRIPTION
**Pull request by viliam-durina**
It had issues:
- single use ill-defined Util.read() method
- improper use of InputStream.available() to detect EOF
- didn't handle empty files (not a requirement, but anyway)

### CODE REVIEW

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ResourceIteratorTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            }
+            assertEquals("offsets not in sequence", lastOffset, rp.getOffset());
+            lastOffset += rp.getBytes().length;
+            assertEquals("contents don't match", contents[fileIndex].substring(rp.getOffset(), lastOffset), new String(rp.getBytes()));
+        }
+        assertEquals("not all config files read", lastIndex, numFiles - 1);
+        assertEquals("config not fully read", contents[lastIndex].length(), lastOffset);
+
+        // delete the files
+        for (Path file : files) {
```
Comment (cangencer) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ResourceIteratorTest.java]: is this necessary? you already have deleteOnExit()

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java [hunk 2]
```diff
@@ -64,33 +64,41 @@ public void remove() {
...
-                ResourcePart part = new ResourcePart(resourceConfig.getDescriptor(), bytes, offset);
-                offset += bytes.length;
-                return part;
-            } else {
-                throw new NoSuchElementException();
+                int bytesRead = inputStream.read(buffer);
+
+                if (bytesRead < 0) {
+                    bytesRead = 0;
+                    inputStream.close();
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java]: is inputStream closed when there is some exception?

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java [hunk 3]
```diff
@@ -20,23 +20,27 @@
...
 
     public ResourceIterator(Set<ResourceConfig> resourceConfigs, int partSize) {
         this.configIterator = resourceConfigs.iterator();
-        this.partSize = partSize;
+
+        buffer = new byte[partSize];
+        readNext();
     }
 
     private void switchFile() throws IOException {
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java]: is the configIterator.hasNext() check still necessary here?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ResourceIteratorTest.java (hunk 1 above)
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ResourceIteratorTest.java]: it's like early delete. If you run all the tests, they will be deleted earlier. I moved it to finally and remove deleteOnExit

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java (hunk 2 above)
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/ResourceIterator.java]: It's not, it's also not, when we don't finish the iteration, if the exception is in our caller.

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Pull request merged**